internal/rest/rest_operation.go

```diff
@@ -270,17 +421,210 @@ func AviRestOperate(c *clients.AviClient, rest_ops []*utils.RestOp) error {
+					refreshCacheForRetry := false
+					if avimodel != nil && isEvh && len(avimodel.GetAviEvhVS()) > 0 {
+						refreshCacheForRetry = true
+					} else if avimodel != nil && !isEvh && len(avimodel.GetAviVS()) > 0 {
+						refreshCacheForRetry = true
+					}
+					if refreshCacheForRetry {
+						utils.AviLog.Warnf("key: %s, msg: Retrieved key for Retry:%s, object: %s", key, publishKey, rest_ops[i].ObjName)
+						aviError, ok := rest_ops[i].Err.(session.AviError)
+						if !ok {
+							utils.AviLog.Infof("key: %s, msg: Error is not of type AviError, err: %v, %T", key, rest_ops[i].Err, rest_ops[i].Err)
+							continue
+						}
+						retryable, fastRetryable, nextObj := f.restOp.RefreshCacheForRetryLayer(publishKey, aviObjKey, rest_ops[i], aviError, aviclient, avimodel, key, isEvh)
+						retry = retry || retryable
+						processNextObj = processNextObj || nextObj
+						if avimodel.GetRetryCounter() != 0 {
+							fastRetry = fastRetry || fastRetryable
+						} else {
+							fastRetry = false
+							utils.AviLog.Warnf("key: %s, msg: retry count exhausted, would be added to slow retry queue", key)
+						}
+					} else {
+						utils.AviLog.Warnf("key: %s, msg: Avi model not set, possibly a DELETE call", key)
+						aviError, ok := rest_ops[i].Err.(session.AviError)
+						// If it's 404, don't retry
+						if ok {
+							statuscode := aviError.HttpStatusCode
+							if statuscode != 404 {
+								f.restOp.PublishKeyToSlowRetryLayer(publishKey, key)
+								//Here as it is 404 for specific object in a current child, AKO can go ahead with next child
+								return false, true
+							} else {
+								f.restOp.AviVsCacheDel(rest_ops[i], aviObjKey, key)
+							}
+						}
+					}
+				} else {
+					f.restOp.PopulateOneCache(rest_ops[i], aviObjKey, key)
+				}
+			}
+
+			if retry {
+				if fastRetry {
+					f.restOp.PublishKeyToRetryLayer(publishKey, key)
+				} else {
+					f.restOp.PublishKeyToSlowRetryLayer(publishKey, key)
+				}
+			}
+			return false, processNextObj
+		}
+	}
+	return true, true
+}
+
+func (f *follower) AviRestOperateWrapper(aviClient *clients.AviClient, rest_ops []*utils.RestOp, key string) error {
+	restTimeoutChan := make(chan error, 1)
+	go func() {
+		err := f.AviRestOperate(aviClient, rest_ops, key)
+		restTimeoutChan <- err
+	}()
+	select {
+	case err := <-restTimeoutChan:
+		return err
+	case <-time.After(lib.ControllerReqWaitTime * time.Second):
+		utils.AviLog.Warnf("key: %s, msg: timed out waiting for rest response after %d seconds", key, lib.ControllerReqWaitTime)
+		return errors.New("timed out waiting for rest response")
+	}
+}
+
+func (f *follower) AviRestOperate(c *clients.AviClient, rest_ops []*utils.RestOp, key string) error {
+	for i, op := range rest_ops {
+		SetTenant := session.SetTenant(op.Tenant)
+		SetTenant(c.AviSession)
+		if op.Version != "" {
+			SetVersion := session.SetVersion(op.Version)
+			SetVersion(c.AviSession)
+		}
+		op.Path += "?name=" + op.ObjName
```

Comment:
<comment from code review> createdby, tenant, cloud must be appended since the name may not be unique.

Reply:
Addressed.